drivers: Update MCUX I2C RTIO driver with SDK changes

ubieda opened the pull request:
### Description
Update the driver to account for variations in the SDK driver
when it uses the instance number instead of the base address.

Changes come from #70866 and #83704 applied to I2C RTIO.

Fixes #88319.

### Testing

Built for MCXN947 and tested I2C commands work.

```
west build -b frdm_mcxn947/mcxn947/cpu0 samples/subsys/shell/shell_module -- -DCONFIG_I2C=y -DCONFIG_I2C_SHELL=y -DCONFIG_I2C_RTIO=y
```

Output:
```
uart:~$ i2c scan lpi2c@
  lpi2c@b7800  lpi2c@94800  
uart:~$ i2c scan lpi2c@94800 
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:             -- -- -- -- -- -- -- -- -- -- -- -- 
10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
50: -- -- -- 53 -- -- -- -- -- -- -- -- -- -- -- -- 
60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
70: -- -- -- -- -- -- -- --                         
1 devices found on lpi2c@94800
uart:~$ 
```

ubieda committed:
drivers: i2c_rtio: Use MCUX variations when using instance number

Update the driver to account for variations in the SDK driver
when it uses the instance number instead of the base address.

Applying 49bdcd2ef2fd5c91ab36f08d29e5183df5437843 on RTIO-version.

Co-authored-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>
Signed-off-by: Luis Ubieda <luisf@croxel.com>

ubieda committed:
drivers: i2c_rtio: Use flexcomm interface to manage interrupts

Some NXP SoC's have a FlexComm interface that manages the
interrupts.

Applying 482e39ea9556f53adbb7f67d0d0da3d17bbbae90 on RTIO-version.

Authored-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>
Signed-off-by: Luis Ubieda <luisf@croxel.com>

ubieda committed:
drivers: i2c_rtio: Use the NXP Flexcomm driver for interrupt handling

The Low Power Flexcomm driver manages the interrupt handling
and provides an API to register interrupt callbacks.
Register the NXP LPI2C interrupt handler.

Applying dca6e64c93f26db254089f20225854bb1f8fe9b4 on RTIO-version.

Authored-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>
Signed-off-by: Luis Ubieda <luisf@croxel.com>

decsny reviewed (changes requested):
this PR is reverting https://github.com/zephyrproject-rtos/zephyr/pull/83704 , I don't understand the point

ubieda commented:
> this PR is reverting #83704 , I don't understand the point

@decsny I'm confused, please ellaborate:
- The point is that without these changes I2C RTIO does not work for LPI2C (at least on MCXN947) at all.
- The referenced PR does not modify `i2c_mcux_lpi2c_rtio.c`. I don't see how this is a revert.
- If the NACK is about including the changes in the referenced PR. I'm happy to do so.

Please provide feedback so we're on the same page before iterating.

decsny commented:
> > this PR is reverting #83704 , I don't understand the point
> 
> @decsny I'm confused, please ellaborate:
> 
>     * The point is that without these changes I2C RTIO does not work for LPI2C (at least on MCXN947) at all.
> 
>     * The referenced PR does not modify `i2c_mcux_lpi2c_rtio.c`. I don't see how this is a revert.
> 
>     * If the NACK is about including the changes in the referenced PR. I'm happy to do so.
> 
> 
> Please provide feedback so we're on the same page before iterating.

I see, I confused them for the same file. In that case please follow the same paradigm as is being done for the other file. (or if you're feeling really ambitious, make some common file to share this, but I don't suggest that for this PR)

ubieda committed:
drivers: i2c_rtio: MCUX LPflexcomm determine by inst

Determine if lpflexcomm wrapped lpi2c by instance and connect
irq differently dependending on that to support platforms with
both flexcomm wrapped and unwrapped lpi2c's.

Applying c1286a8d8d425805fcceb3b872325fb4c439a572 to RTIO version.

Authored-by: Declan Snyder <declan.snyder@nxp.com>
Signed-off-by: Luis Ubieda <luisf@croxel.com>

teburd commented:
just ran into this myself, thanks for fixing
